pkg/monitortests/kubeapiserver/staticpodinstall/kubeletlogparser/parser.go

```diff
@@ -0,0 +1,51 @@
+package kubeletlogparser
+
+import (
+	"strings"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+)
+
+// NewEtcdStaticPodEventsFromKubelet returns a parser that will parse kubelet
+// log for SyncLoop PLEG and probe events.
+// For now, we are interested in the etcd installer and the etcd static pod
+func NewEtcdStaticPodEventsFromKubelet() func(node, line string) monitorapi.Intervals {
+	filter := func(node, ns, podName string) bool {
+		return ns == "openshift-etcd" && (strings.HasPrefix(podName, "installer-") || podName == "etcd-"+node)
+	}
+	p := parsers{
+		// we want to observe the unready window for ectd static pods
+		&SyncLoopProbeParser{
+			source: monitorapi.SourceKubeletLog,
+			want:   filter,
+		},
+		// we want to observe the container start and exit pleg events for the etcd installer pods
+		&SyncLoopPLEGParser{
+			source: monitorapi.SourceKubeletLog,
+			filter: filter,
+		},
+	}
+	return p.parse
+}
+
+// interval is created only if the given filter func returns true
+type filterFunc func(node, ns, podName string) bool
+
+type parser interface {
+	Parse(node, line string) (monitorapi.Intervals, bool)
+}
+
+type parsers []parser
+
+// this complies with the signature the default node log parser in origin requires
+func (p parsers) parse(node, line string) monitorapi.Intervals {
+	accummulated := monitorapi.Intervals{}
+	for _, parser := range p {
+		intervals, handled := parser.Parse(node, line)
+		accummulated = append(accummulated, intervals...)
```

Comment:
What if (in the future) either of Parse implementations accidentally returns non-nil intervals and `handled=false`? Will accummulated be still a valid list of intervals?

Reply:
we can tweak it in the future, forcefully run all parsers, we also want to streamline all the existing  parsers - https://github.com/openshift/origin/blob/5b828df7f1c6648b6384170f0e5dd0614aee3e59/pkg/monitortests/node/kubeletlogcollector/node.go#L96-L106

I had to move the two parsers in this PR n their own packages so I could write some tests. to your question, right now, if that happens the existing tests will fail, so we have some protection :)